meson.build

```diff
@@ -147,15 +129,23 @@ deps += sdl2_ttf_dep
 # If SDL2_mixer was found without using the fallback, chances are that it already
 # has all its dependencies.
 if freetype_use_embedded
-  freetype2_dep = dependency('freetype2', required: false)
+  freetype2_dep = dependency(
+    'freetype2',
+    required: false,
+    default_options: ['png=disabled', 'zlib=disabled']
+    )
   if not freetype2_dep.found()
-    freetype2_dep = cc.find_library('freetype2', has_headers: 'freetype.h', required: false)
+    freetype2_dep = cc.find_library(
+      'freetype2',
+      has_headers: 'freetype.h',
+      required: false
+      )
     if not freetype2_dep.found()
       freetype2_dep = dependency(
         'freetype2',
         required: true,
         fallback: 'freetype2',
-        default_options: ['default_library=static'])
+        default_options: ['default_library=static', 'png=disabled', 'zlib=disabled'])
```

Comment:
@eli-schwartz Can you tell me why this has absolutely no effect?

```
freetype2 

  Configuration Options Summary:
    OS      : linux
    Zlib    : yes
    Bzip2   : no
    Png     : yes
    Harfbuzz: no
    Brotli  : no

physfs 3.2.0

  Archive support
    zip    : enabled
    7z     : enabled
    grp    : disabled
    wad    : disabled
    hog    : disabled
    mvl    : disabled
    qpak   : disabled
    slb    : disabled
    iso9660: disabled
    vdf    : disabled

netpanzer 0.9.0.07.555

  Subprojects
    freetype2 : YES 8 warnings
    libpng    : YES
    lua       : YES
    ogg       : YES
    physfs    : YES
    sdl2      : YES
    sdl2_mixer: YES
    sdl2_ttf  : YES 1 warnings
    vorbis    : YES
    zlib      : YES
```

Reply:
Most often that happens when you add this after the fact, and your existing build directory has been configured with different settings. The default options only kick in when newly setting up the build directory (they are only defaults, after all).

Reply:
I saw that in the docs, but that isn't the case. I run `meson setup _build` after making the changes. I've done it several times. I'm using meson v1.3.1 installed from pip on Ubuntu Jammy

Reply:
Several times... either after removing the _build directory or using a different name. ;)

Reply:
Then probably another dependency that is found earlier, is internally using `dependency(freetype2')` and causing it to be configured without any options. Try looking for freetype first.